### PR TITLE
[Main Course] - sql tips

### PR DESCRIPTION
**Pull request by jhqian0909**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Post for sql tips.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+```
+where->join->group by->qualify->select
+```
+
+这边和Hive不同的是，Hive的多表join在where之前，同时Hive会把左表载入内存，所以我们需要先对每个表做筛选再join来优化查询。  
+关于Teradata的优化，我们要做的是最小化I/O，确保每个步骤过滤掉尽可能多的数据。  
+
+Snowflake是比较新的数据库，公司刚上市不久，之前所述的数据库都有几十年的历史了。它的特点是云数据库，在万物上云的今天，上云已经成为了一种必须。  
+除了它的技术架构比较新颖，对于用户体验来说，Snowflake有如下优点：  
+- 功能繁多(一些EDA的图表等)，美观的UI。
```
Comment (Realvincentyuan) [inline docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md:35]: 要空行

File: docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+- 功能强大的查询语言，如支持qualify等，或者说没什么缺陷和硬伤。
+
+## 4 Hadoop
+
+在Hadoop生态圈中，用的较多是Hive和Spark。Hive是基于Hadoop的一个数据仓库工具，可以用Hive SQL执行MapReduce操作。目前大规模的数据存储(国内外互联网)感觉还是以Hadoop为主。  
+在工作中也用了一段时间的Hive了，这里总结下踩过的坑。
+- Hive的子查询一定要有别名。
```
Comment (Realvincentyuan) [inline docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md:42]: 这些不同的元素之间应该都要空行

File: docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  ```sql
+  select * from 
+  (select * from table) a
+  ```
+- Hive中的limit会在所有逻辑执行完后才limit，可以使用如下设置优化：
```
Comment (Realvincentyuan) [inline docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md:47]: 这里是优化什么啊？

**Review by Realvincentyuan — APPROVED** [review]
有些小评论

File: docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md (hunk 3 above)
Comment (jhqian0909) [reply docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md]: 会对数据源进行抽样

File: docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md (hunk 2 above)
Comment (jhqian0909) [reply docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md:44]: 已修改

File: docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md (hunk 1 above)
Comment (jhqian0909) [reply docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md:36]: 已修改

File: docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md (hunk 3 above)
Comment (Realvincentyuan) [reply docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md]: 在最开始取子表的时候就进行抽样了吗？

File: docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  ```
+- Hive传递参数：
+
+  ```sql
+  set hivevar:var=1;
+  select ${var}
+  其中，若参数本身是字符串还需要加上引号，如date_sub('${timestamp}',7)
+  ```
+  
+- 数据是否倾斜会极大的影响查询效率
```
Comment (jz263) [inline docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md:63]: 这里的数据倾斜的定义可能要找为解释一下

File: docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md (hunk 3 above)
Comment (jhqian0909) [reply docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md]: 我理解是的。从最开始的表里就抽样了。

File: docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md (hunk 4 above)
Comment (jhqian0909) [reply docs/Main_Course/Data_Analytics/2021-08-18-Sql-tips.md]: 已修改

**Pull request merged**